pkg/transport/session/manager.go

```diff
@@ -113,13 +124,15 @@ func (m *Manager) AddWithID(id string) error {
 	if id == "" {
 		return fmt.Errorf("session ID cannot be empty")
 	}
-	// Use LoadOrStore: returns existing if already present
-	session := m.factory(id)
-	_, loaded := m.sessions.LoadOrStore(id, session)
-	if loaded {
+	// Check if session already exists
+	ctx := context.Background()
```

Comment:
why do we have some contexts with timeouts and others without?

Reply:
Good catch! I've fixed this inconsistency. Now all storage operations use consistent context timeouts:

- **5 seconds** for quick operations: `Get()`, `Delete()`, `AddWithID()`, `AddSession()`
- **30 seconds** for cleanup operations: `DeleteExpired()` in both the cleanup routine and `cleanupExpiredOnce()`

This provides:
1. Consistent behavior across the codebase
2. Protection against hanging operations
3. Reasonable timeouts for local storage (should be near-instant) while allowing buffer for future distributed storage backends